### ISSUE TITLE
Use crate-ci/typos to catch typos in CI

### ISSUE THREAD

**Issue by leoloso** [GatoGraphQL/GatoGraphQL#2726]
Integrate the monorepo with https://github.com/crate-ci/typos, to catch typos before merging a PR.

The following `typos.toml` file (thanks @szepeviktor) is a good start, but it needs to be improved, as there are a few false positives:

```toml
[files]
extend-exclude = [
    ".git/",
    "**/vendors/**",
    "**/build/**",
    "**.wordpress.xml",
    "**/gatographql-testing-data.xml",
    "**/fixture-schema/*.json",
    "layers/GatoGraphQLForWP/plugins/gatographql/extensions/http-request-via-schema/docs/modules/http-request-via-schema/en.md",
    "layers/GatoGraphQLForWP/plugins/gatographql/docs/tutorial/bulk-translating-block-content-in-multiple-posts-to-a-different-language/en.md",
    "layers/GatoGraphQLForWP/plugins/gatographql/docs/tutorial/translating-block-content-in-a-post-to-a-different-language/en.md",
    "layers/GatoGraphQLForWP/plugins/gatographql/docs/tutorial/translating-content-from-url/en.md",
    "layers/GatoGraphQLForWP/plugins/gatographql/docs/release-notes/1.2/en.md",
    "layers/GatoGraphQLForWP/phpunit-packages/gatographql/tests/Integration/fixture-admin-client/introspection-query.json",
    "layers/GatoGraphQLForWP/phpunit-packages/gatographql/tests/Integration/fixture-schema-mutations/success/media-contents:1.var.json",
    "layers/GatoGraphQLForWP/phpunit-packages/gatographql/tests/Integration/fixture-queryable-customposts/queryable-customposts:2.json",
    "layers/GatoGraphQLForWP/phpunit-packages/gatographql/tests/Integration/fixture-queryable-customposts/queryable-customposts:0.json",
    "layers/GatoGraphQLForWP/phpunit-packages/wpfaker-schema/resources/_fixed-dataset.wordpress.php",
    "layers/GatoGraphQLForWP/plugins/gatographql/docs/release-notes/1.2/en.md",
    "layers/GatoGraphQLForWP/plugins/gatographql/docs/tutorial/translating-content-from-url/en.md",
    "layers/GatoGraphQLForWP/plugins/gatographql/extensions/google-translate/docs/modules/google-translate/en.md",
    "layers/GatoGraphQLForWP/plugins/gatographql/setup/persisted-queries/admin/sync/import-posts-from-csv.gql",
    "**/input-object-inner-validations.json",
]
ignore-hidden = false

[default]
extend-ignore-re = [
    "And gettin' caught",
    "Allow URE",
    "URE source",
    "X-Github-Access-Token",
]

[default.extend-identifiers]
safly = "safly"
marketing_optin = "marketing_optin"
hasSubscriberOptin = "hasSubscriberOptin"
subscriberOptin = "subscriberOptin"
isNotSubscriberOptinNAValue = "isNotSubscriberOptinNAValue"
QueryRoot2ndLevel = "QueryRoot2ndLevel"
inputOn2ndLevel = "inputOn2ndLevel"
no2ndLevelInput = "no2ndLevelInput"

# Typos
"Github" = "GitHub"
"Wordpress" = "WordPress"
```

**Comment by szepeviktor** [comment]
You could remove all "optin" variables and add
```toml
[default.extend-words]
optin = "optin"
```

**Comment by leoloso** [comment]
Can also rename those variables. Data is sent as "optin" in the InstaWP webhook, but it can be renamed to something more proper on my side.

**Comment by szepeviktor** [comment]
Actually
`hasSubscriberOptin` is
`hasSubscriberOptIn`.

**Comment by leoloso** [comment]
> Actually `hasSubscriberOptin` is `hasSubscriberOptIn`.
> 
> https://www.merriam-webster.com/dictionary/opt%20in

Ok, fixed in [c5a0419](https://github.com/GatoGraphQL/GatoGraphQL/commit/c5a04191b00a24ed614ccfa2d5fffd2baf0e44d6)

**Comment by szepeviktor** [comment]
The workflow.
```yaml
name: Spelling
on:
    push:
        branches:
            - master
    pull_request: null
env:
    CHECKOUT_SUBMODULES: ""

jobs:
    main:
        name: Typos
        runs-on: ubuntu-latest
        steps:
            -   name: Checkout code
                uses: actions/checkout@v4
                with:
                    submodules: ${{ env.CHECKOUT_SUBMODULES }}

            -   name: Search for misspellings
                uses: crate-ci/typos@master
```

**Comment by leoloso** [comment]
Implemented in #2729
